rainier-core/src/main/scala/com/stripe/rainier/compute/Translator.scala

```diff
@@ -59,9 +59,15 @@ private class Translator {
     SeqIR(defs :+ lookupExpr)
   }
 
+  private def sumExpr(exprs: List[Expr]): Expr = {
+    val defs = exprs.collect { case v: VarDef => v }
+    val sumExpr = VarDef(SumIR(exprs))
```

Comment:
You need to get `ref`s of all the exprs to stick in the SumIR rather than referencing the identical nodes twice.